### PR TITLE
demo: update Progress demo

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

加了两个滑动条

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | demo: update Progress demo |
| 🇨🇳 Chinese | demo: update Progress demo |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/progress/demo/circle-steps.tsx [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
-    <Progress
-      type="circle"
-      percent={100}
-      steps={{ count: 5, gap: 12 }}
-      trailColor="rgba(0, 0, 0, 0.06)"
-      strokeWidth={20}
-    />
-  </Space>
-);
+const DEFAULT_COUNT = 5;
```
Comment (crazyair) [inline components/progress/demo/circle-steps.tsx:4]: 直接写到 useState 里呗，不用单独提出来吧

Comment (li-jia-nan) [reply components/progress/demo/circle-steps.tsx:4]: 这个复用了几次，我就拿出来了

File: components/progress/demo/circle-steps.tsx [hunk 2]
```diff
@@ -1,23 +1,49 @@
...
+const DEFAULT_GAP = 8;
+
+const App: React.FC = () => {
+  const [stepsCount, setStepsCount] = React.useState<number>(DEFAULT_COUNT);
+  const [stepsGap, setStepsGap] = React.useState<number>(DEFAULT_GAP);
+  return (
+    <>
+      <p>Custom count:</p>
+      <Slider
+        defaultValue={DEFAULT_COUNT}
```
Comment (crazyair) [inline components/progress/demo/circle-steps.tsx:14]: 这个 props 可以不传，因为已经传了 value 了，这个其实也没作用了

Comment (li-jia-nan) [reply components/progress/demo/circle-steps.tsx:14]: 有道理

File: components/progress/demo/circle-steps.tsx [hunk 3]
```diff
@@ -1,23 +1,49 @@
...
+        min={2}
+        max={10}
+        value={stepsCount}
+        onChange={setStepsCount}
+      />
+      <p>Custom gap:</p>
+      <Slider
+        defaultValue={DEFAULT_GAP}
```
Comment (afc163) [inline components/progress/demo/circle-steps.tsx:22]: defaultValue 和 value 不应该同时使用。

File: components/progress/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 4]
```diff
@@ -622,7 +710,7 @@ exports[`renders components/progress/demo/circle-steps.tsx extend context correc
             opacity="1"
             r="40"
             stroke-width="20"
-            style="stroke-dasharray: 251.32741228718345px 251.32741228718345; stroke-dashoffset: 213.06192982974676; transform: rotate(-18deg); transform-origin: 50px 50px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
+            style="stroke-dasharray: 251.32741228718345px 251.32741228718345; stroke-dashoffset: 201.06192982974676; transform: rotate(NaNdeg); transform-origin: 50px 50px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s; fill-opacity: 0; transition-duration: 0s, 0s;"
```
Comment (crazyair) [inline components/progress/__tests__/__snapshots__/demo-extend.test.ts.snap:713]: NaN

Comment (li-jia-nan) [reply components/progress/__tests__/__snapshots__/demo-extend.test.ts.snap:713]: 奇怪

File: components/progress/demo/circle-steps.tsx [hunk 5]
```diff
@@ -1,23 +1,33 @@
...
-      trailColor="rgba(0, 0, 0, 0.06)"
-      strokeWidth={20}
-    />
-  </Space>
-);
+const App: React.FC = () => {
+  const [stepsCount, setStepsCount] = React.useState<number>(5);
+  const [stepsGap, setStepsGap] = React.useState<number>(7);
+  return (
+    <>
```
Comment (zombieJ) [inline components/progress/demo/circle-steps.tsx:8]: 感觉 Flex + Typography.Title level=5 会比较好，不应直接出现 p 元素。

Comment (li-jia-nan) [reply components/progress/demo/circle-steps.tsx:8]: done

**Review by zombieJ — APPROVED** [review]

**Review by crazyair — APPROVED** [review]

**Pull request merged**